Modix/Modules/NoteModule.cs

```diff
@@ -0,0 +1,200 @@
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"NoteModule RemoveNote failed with the following noteId: {noteId}");
+                await ReplyAsync($"Failed to remove the note: {noteId}");
+            }
+        }
+
+        [Command("note search"), Summary("Search notes for a user")]
+        public async Task SearchNotesByUserId(ulong userId)
+        {
+            try
+            {
+                var notes = _noteRetrieverService.FindNotesForUser(userId);
+
+                if (notes == null)
+                {
+                    Log.Error("Notes has thrown an error");
+                    await ReplyAsync($"No notes found for user {userId}");
+                }
+
+
+                if (notes.Count == 0)
+                {
+                    await ReplyAsync($"No notes found for user {userId}");
+                    return;
+                }
+
+                var sb = new StringBuilder();
+                var hints = new Hints { MaxTableWidth = 100 };
+                var formatter = new TableFormatter(hints);
+                sb.AppendLine("```");
+
+                var formattedNotes = notes.Select(note => new
+                {
+                    Id = $"{note.Id}",
+                    User = $"{note.UserName}",
+                    RecordedBy = $"{note.RecordedBy}",
+                    Message = $"{note.Message}",
+                    Date = $"{note.RecordedTime}"
+                }).ToList();
+
+                var text = formatter.FormatObjects(formattedNotes);
+                sb.Append(text);
+
+                sb.AppendLine("```");
+
+                if (sb.ToString().Length <= 2000)
+                {
+                    await ReplyAsync(sb.ToString());
+                    return;
+                }
+
+                var formattedNoteIds = notes.Select(note => new
+                {
+                    NoteId = $"{note.Id}"
+                });
+
+                var noteIds = formatter.FormatObjects(formattedNoteIds);
+
+                sb.Clear();
+                sb.AppendLine("Notes exceed the character limit. Search for an Id below to retrieve note details");
+                sb.AppendLine("```");
+                sb.Append(noteIds);
+                sb.AppendLine("```");
+
+                await ReplyAsync(sb.ToString());
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"NoteModule SearchNotesByUserId failed with the following userId: {userId}");
+                await ReplyAsync("Error occurred and search could not be complete");
+            }
+        }
+
+        [Command("note search"), Summary("Search notes for a username")]
+        public async Task SearchNotesByUserName(IGuildUser user)
+            => await SearchNotesByUserId(user.Id);
+
+        [Command("note search id"), Summary("Search notes for a user")]
+        public async Task SearchNotesByNoteId(int noteId)
```

Comment:
It looks like the summary needs updated for this. But is this command necessary? Seems odd to want to retrieve a note directly by the note's ID.

Reply:
This strikes me as a useful method on `INoteService`, for use by the webapp, but not for a command, yeah.